__tests__/commands/upgrade.js

```diff
@@ -104,6 +104,24 @@ test.concurrent('upgrades from fixed version to latest', (): Promise<void> => {
   });
 });
 
+test.concurrent('upgrades package not in registry', (): Promise<void> => {
+  return runUpgrade(['yarn-test-git-repo'], {}, 'package-not-in-registry', async (config): ?Promise<void> => {
+    const lockfile = explodeLockfile(await fs.readFile(path.join(config.cwd, 'yarn.lock')));
+    const gitRemote = 'https://github.com/juanca/e2e-test-repo';
+
+    assert.equal(
+      lockfile[0],
+      `"yarn-test-git-repo@${gitRemote}#package-not-in-registry":`,
+      'Lockfile should point to the same branch.',
+    );
+
+    assert(
+      lockfile.indexOf(`${gitRemote}#d2027157d0c7188fc9ed6a6654325d1e3bf4db40`) === -1,
```

Comment:
Welp, I'm really glad I'm adding multiple packages to this test. This particular assertion is actually not doing anything. As with any test asserting the non-existence of something, using the same assertion for the existence of something is required. This will shortly be fixed.